src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseConcreteTypeAnalyzer.cs

```diff
@@ -323,6 +360,37 @@ void Evaluate(ISymbol affectedSymbol, ITypeSymbol fromType, PooledConcurrentSet<
                 reportDiag(diagnostic);
             }
 
+            // ensures that the type can be referenced from any code that can also reference the symbol
+            bool HasEquivalentOrGreaterVisibilityToSymbol(ITypeSymbol type, ISymbol affectedSymbol)
```

Comment:
Nit: pass `compilation` as an argument and make this a static local function.

Reply:
Done.